File: src/main/java/io/vproxy/vfx/ui/button/AbstractFusionButton.java
package io.vproxy.vfx.ui.button;

import io.vproxy.vfx.theme.Theme;
import io.vproxy.vfx.ui.pane.AbstractFusionPane;
import javafx.scene.paint.Color;

public abstract class AbstractFusionButton extends AbstractFusionPane {
    protected abstract void onMouseClicked();

    @Override
    protected Color normalColor() {
        return Theme.current().fusionButtonNormalBackgroundColor();
    }

    @Override
    protected Color hoverColor() {
        return Theme.current().fusionButtonHoverBackgroundColor();
    }

    @Override
    protected Color downColor() {
        return Theme.current().fusionButtonDownBackgroundColor();
    }
}

File: src-ts/src/ui/button/AbstractFusionButton.ts
import { Color } from '../../javafx/color.js';
import { Theme } from '../../theme/Theme.js';
import { AbstractFusionPane } from '../pane/AbstractFusionPane.js';

export abstract class AbstractFusionButton extends AbstractFusionPane {
  protected abstract override onMouseClicked(): void;

  protected override normalColor(): Color {
    return Theme.current().fusionButtonNormalBackgroundColor();
  }

  protected override hoverColor(): Color {
    return Theme.current().fusionButtonHoverBackgroundColor();
  }

  protected override downColor(): Color {
    return Theme.current().fusionButtonDownBackgroundColor();
  }
}
